src/test/java/TestUtils.java

```diff
@@ -29,27 +29,47 @@ private TestUtils() {
+      BlobKey blobkey, String imageUrl, double price, String store, ImmutableSet<String> categories,
+      String rawText) {
+    Entity receiptEntity =
+        createEntity(userId, timestamp, blobkey, imageUrl, price, store, categories, rawText);
+
+    datastore.put(receiptEntity);
+    return receiptEntity;
+  }
+
+  /** Adds multiple receipts to datastore. */
   public static ImmutableSet<Entity> addTestReceipts(DatastoreService datastore) {
-    ImmutableSet<Entity> testReceipts = ImmutableSet.of(
-        addTestReceipt(datastore, /* userId = */ "123", /* timestamp = */ 1045237591000L,
-            new BlobKey("test"), "img/walmart-receipt.jpg", 26.12, "walmart",
-            ImmutableSet.of("candy", "drink", "personal"), ""),
-
-        addTestReceipt(datastore, /* userId = */ "123", /* timestamp = */ 1560193140000L,
-            new BlobKey("test"), "img/contoso-receipt.jpg", 14.51, "contoso",
-            ImmutableSet.of("cappuccino", "sandwich", "lunch"), ""),
-
-        addTestReceipt(datastore, /* userId = */ "123", /* timestamp = */ 1491582960000L,
-            new BlobKey("test"), "img/restaurant-receipt.jpeg", 29.01, "main street restaurant",
-            ImmutableSet.of("food", "meal", "lunch"), ""));
+    ImmutableSet<Entity> testReceipts = createEntities();
+
+    testReceipts.stream().forEach(receipt -> datastore.put(receipt));
+
     return testReceipts;
   }
 
-  /** Adds a test receipt to the mock datastore and returns the id of that entity. */
-  public static Entity addTestReceipt(DatastoreService datastore, String userId, long timestamp,
-      BlobKey blobkey, String imageUrl, double price, String store, ImmutableSet<String> categories,
-      String rawText) {
+  /** Create multiple entities to be added to datastore. */
+  public static ImmutableSet<Entity> createEntities() {
+    ImmutableSet<Entity> entities =
+        ImmutableSet.of(createEntity(/* userId = */ "123", /* timestamp = */ 1045237591000L,
+                            new BlobKey("test"), "img/walmart-receipt.jpg", 26.12, "walmart",
+                            ImmutableSet.of("candy", "drink", "personal"), ""),
+
+            createEntity(/* userId = */ "123", /* timestamp = */ 1560193140000L,
+                new BlobKey("test"), "img/contoso-receipt.jpg", 14.51, "contoso",
+                ImmutableSet.of("cappuccino", "sandwich", "lunch"), ""),
+
+            createEntity(/* userId = */ "123", /* timestamp = */ 1491582960000L,
+                new BlobKey("test"), "img/restaurant-receipt.jpeg", 29.01, "main street restaurant",
+                ImmutableSet.of("food", "meal", "lunch"), ""));
+
+    return entities;
+  }
+
+  /** Creates and returns a single Receipt entity. */
+  public static Entity createEntity(String userId, long timestamp, BlobKey blobkey, String imageUrl,
+      Double price, String store, ImmutableSet<String> categories, String rawText) {
```

Comment:
Is there a reason for using the `Double` wrapper class instead of `double` here?

Reply:
It looks like a typo given that the createEntity calls all use a `double`.

Reply:
In #87 there's a test where price is null. The Double wrapper here will allow for this null check